gui/src/components/Txs.tsx

```diff
@@ -88,18 +91,18 @@ function Receipt({ hash }: { hash: string }) {
   );
 }
 
-function Status({ status }: { status?: number }) {
+function Status({ status }: { status: TransactionReceipt["status"] }) {
```

Comment:
@naps62 TS was complaining here because `number` isn't compatible with the type `success | reverted` coming from `TrandactionReceipt` - Did I miss something?

Reply:
I'm not sure about this. on the surface. sounds like your code is correct, but I'm now wondering why it was working before

Reply:
Maybe because `status` is typed differently with Ethers?

<img width="712" alt="Screenshot 2023-04-13 at 14 29 39" src="https://user-images.githubusercontent.com/26620750/231774002-2ca8dfd0-0fe9-4fc4-a975-ee19156fdc80.png">

https://docs.ethers.org/v5/api/providers/types/#:~:text=receipt.status%20%E2%87%92%20number
vs. Viem

<img width="530" alt="Screenshot 2023-04-13 at 14 30 30" src="https://user-images.githubusercontent.com/26620750/231774282-c8efb556-9568-4191-93a0-e34fe4edfa92.png">

https://github.com/wagmi-dev/viem/blob/0a148a983064a7183f030e63b9de0b3c9a0bc8d6/src/types/transaction.ts#L15

Either way, even if we don't go with Viem, should we be using the Ethers type for status instead of hardcoding `number`?

Reply:
ohhh, they convert that to a string directly.
anyway. I think using `number` is perfectly fine while we stick with ethers
the spirit of typescript is that you don't need to specify the exact type all the time. If you imagine the codebase filled with `TransactionReceipt["xxx"]` instead of directly `string` or `number` or other basic types, it'd get a lot more confusing

changing to viem will cause typescript to show these errors, which (correctly) tells us that we need to change how we handle the value. so I'd say it's all working as intended

Reply:
Okay, I see.

Reply:
Are we not changing to Viem then? 😕

Reply:
we are. I was answering your "even if we don't go with Viem" question

Reply:
@naps62 @gabrielpoca? Can we merge this, then? Or did I miss something that needs fixing still?

Reply:
From what I understood yeah, this look ready, but we might want to do some manual testing before merging, I'll try to give it a go